tests/e2e/linalg_ext_ops/attention_i1_mask.mlir

```diff
@@ -0,0 +1,193 @@
+func.func @i1_representation() {
+  %mask = util.unfoldable_constant dense<[140]> : tensor<1xi8>
+  %casted = flow.tensor.bitcast %mask : tensor<1xi8> -> tensor<2x4xi1>
+  %bar = util.optimization_barrier %casted : tensor<2x4xi1>
+  %tensor_res = flow.tensor.bitcast %bar : tensor<2x4xi1> -> tensor<1xi8>
+  check.expect_eq_const(%tensor_res, dense<[140]> : tensor<1xi8>) : tensor<1xi8>
+  return
+}
+
+func.func @i1_representation_2() {
+  %mask = util.unfoldable_constant dense<[140, 77]> : tensor<2xi8>
+  %casted = flow.tensor.bitcast %mask : tensor<2xi8> -> tensor<2x8xi1>
+  %bar = util.optimization_barrier %casted : tensor<2x8xi1>
+  %tensor_res = flow.tensor.bitcast %bar : tensor<2x8xi1> -> tensor<2xi8>
+  check.expect_eq_const(%tensor_res, dense<[140, 77]> : tensor<2xi8>) : tensor<2xi8>
+  return
+}
+
+func.func @i1_representation_3() {
+  %mask = util.unfoldable_constant dense<[140, 77]> : tensor<2xi8>
+  %casted = flow.tensor.bitcast %mask : tensor<2xi8> -> tensor<4x4xi1>
+  %bar = util.optimization_barrier %casted : tensor<4x4xi1>
+  %tensor_res = flow.tensor.bitcast %bar : tensor<4x4xi1> -> tensor<2xi8>
+  check.expect_eq_const(%tensor_res, dense<[140, 77]> : tensor<2xi8>) : tensor<2xi8>
+  return
+}
+
+func.func @truncate_i1() {
+  %mask = util.unfoldable_constant dense<[1, 1, 0, 0,
+                                          0, 0, 1, 1]> : tensor<8xi8>
+  %nm = tensor.empty() : tensor<8xi1>
+  %truncm = linalg.generic
+  {indexing_maps = [
+    affine_map<(d0) -> (d0)>,
+    affine_map<(d0) -> (d0)>],
+  iterator_types = ["parallel"]}
+  ins(%mask: tensor<8xi8>)
+  outs(%nm: tensor<8xi1>) {
+    ^bb0(%in: i8, %out: i1):
+      %zero = arith.constant 0 : i8
+      %truncated = arith.cmpi "sgt", %in, %zero : i8
+      linalg.yield %truncated : i1
+  } -> tensor<8xi1>
+  %tensor_res = flow.tensor.bitcast %truncm : tensor<8xi1> -> tensor<1xi8>
+  check.expect_eq_const(%tensor_res, dense<[195]> : tensor<1xi8>) : tensor<1xi8>
+  return
+}
+
+func.func @truncate_i1_2() {
```

Comment:
can we find some other places for these i1 mask roundtrips? seems like not really exclusive to attention mlir

Reply:
Any thoughts on moving a bunch of these to `subbyte_types.mlir` which you introduced here https://github.com/iree-org/iree/commit/c80fa3bcabbaffab33281c997979ff84974cd843#diff-2b5d11791b3e9bd5976105d499eee9582229f338d72b3e03ecd9f7f222361f0e

Reply:
including `func.func @i1_representation()` in attention.mlir